addons/mrp/models/mrp_production.py

```diff
@@ -2491,6 +2501,9 @@ def filter_by_attributes(record, product=self.product_id):
                     move_raw.workorder_id = self.workorder_ids.filtered(lambda wo: wo.operation_id == move_raw.operation_id)
             elif not bom_line:
                 moves_to_unlink |= move_raw
+
+        self.unlink_old_bom_moves()
```

Comment:
Instead of this, replace with `moves_to_unlink.product_uom_qty = 0` and let the existing behavior update the component pick transfer (i.e. try manually setting the demand to 0 in the unlocked MO's form view and check the pick transfer). The same is probably needed for byproducts in the post production transferas well (i.e. apply the =0 just before the unlinking) It would be better to leave the existing moves in the transfers with a 0 demand than delete any possible data already inputted (i.e. if people started moving stuff + so it's easier to track that it was changed)